src/backend/oneapi/kernel/iota.hpp

```diff
@@ -95,7 +95,7 @@ void iota(Param<T> out, const af::dim4& sdims) {
 
     try {
         getQueue()
-            .submit([=](sycl::handler& h) {
+            .submit([&](sycl::handler& h) {
```

Comment:
we want this by value here right? otherwise the objects will be deleted when they go out of scope.